tests/launchsecurity/sev.go

```diff
@@ -1,28 +1,251 @@
 
+	k8sv1 "k8s.io/api/core/v1"
 	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
 	"kubevirt.io/client-go/kubecli"
 
 	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
 	"kubevirt.io/kubevirt/tests"
 	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/exec"
 	"kubevirt.io/kubevirt/tests/framework/checks"
 	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
 	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/libwait"
 )
 
-var _ = Describe("[sig-compute]AMD Secure Encrypted Virtualization (SEV)", decorators.SigCompute, func() {
+var _ = Describe("[sig-compute]AMD Secure Encrypted Virtualization (SEV)", decorators.SEV, decorators.SigCompute, func() {
+	const (
+		diskSecret = "qwerty123"
+	)
+
+	newSEVFedora := func(opts ...libvmi.Option) *v1.VirtualMachineInstance {
+		sevOptions := []libvmi.Option{
+			libvmi.WithUefi(false),
+			libvmi.WithSEV(),
+		}
+		opts = append(sevOptions, opts...)
+		return libvmi.NewFedora(opts...)
+	}
+
+	// As per section 6.5 LAUNCH_MEASURE of the AMD SEV specification the launch
+	// measurement is calculated as:
+	//   HMAC(0x04 || API_MAJOR || API_MINOR || BUILD ||
+	//        GCTX.POLICY || GCTX.LD || MNONCE; GCTX.TIK)
+	// The implementation is based on
+	//   https://blog.hansenpartnership.com/wp-uploads/2020/12/sevsecret.txt
+	verifyMeasurement := func(sevMeasurementInfo *v1.SEVMeasurementInfo, tikBase64 string) []byte {
+		By("Verifying launch measurement")
+		tik, err := base64.StdEncoding.DecodeString(tikBase64)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		h := hmac.New(sha256.New, tik)
+		err = binary.Write(h, binary.LittleEndian, uint8(0x04))
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		err = binary.Write(h, binary.LittleEndian, uint8(sevMeasurementInfo.APIMajor))
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		err = binary.Write(h, binary.LittleEndian, uint8(sevMeasurementInfo.APIMinor))
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		err = binary.Write(h, binary.LittleEndian, uint8(sevMeasurementInfo.BuildID))
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		err = binary.Write(h, binary.LittleEndian, uint32(sevMeasurementInfo.Policy))
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		loaderSHA, err := hex.DecodeString(sevMeasurementInfo.LoaderSHA)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		h.Write(loaderSHA)
+		m, err := base64.StdEncoding.DecodeString(sevMeasurementInfo.Measurement)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		nonce := m[32:48]
+		h.Write(nonce)
+		measure := m[0:32]
+		ExpectWithOffset(1, hex.EncodeToString(h.Sum(nil))).To(Equal(hex.EncodeToString(measure)))
+		return measure
+	}
+
+	// The implementation is based on
+	//   https://blog.hansenpartnership.com/wp-uploads/2020/12/sevsecret.txt
+	encryptSecret := func(diskSecret string, measure []byte, tikBase64, tekBase64 string) *v1.SEVSecretOptions {
```

Comment:
@vasiliy-ul can be this replaced by the [sevctl secret build](https://github.com/virtee/sevctl#secret-build)

Reply:
That would probably make sense. I will take a look on how to use sevctl here.

Reply:
> > but we could also use it in the node-labeller instead of relying on libvirt endpoints
> 
> For that, we would need to include sevctl into vit-launcher image. Not sure that an administrative tool should be part of the workload container.

Right, my hope is that one day we won't use virt-launcher image in the privileged pod, but this is another story. So, I agree.

Reply:
@vasiliy-ul Which version of sevctl is currently available?

Reply:
@tylerfanelli, its `sevctl-0.3.2-1.el9.x86_64.rpm` from centos 9 stream.

Reply:
@vasiliy-ul we can leave it as a todo